src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataMetadataSerializer.cs

```diff
@@ -34,5 +35,22 @@ public override void WriteObject(object graph, Type type, ODataMessageWriter mes
             // the model received by this method and the model passed(from configuration) while building ODataMessageWriter is the same (clr object).
             messageWriter.WriteMetadataDocument();
         }
+
+        /// <inheritdoc/>
+        /// <remarks>The metadata written is from the model set on the <paramref name="messageWriter"/>. The <paramref name="graph" />
+        /// is not used.</remarks>
+        public override async Task WriteObjectAsync(object graph, Type type, ODataMessageWriter messageWriter, ODataSerializerContext writeContext)
```

Comment:
> public override async Task WriteObjectAsync [](start = 7, length = 44)

shall we use the base "virtual" default implementation?

Reply:
We could, but I like having this as a placeholder for when we implement the WriteMetadataDocumentAsync() method in ODL.

---
In reply to: [515435490](https://github.com/OData/WebApi/pull/2336#discussion_r515435490) [](ancestors = 515435490)